yaksh/views.py

```diff
@@ -1730,6 +1739,29 @@ def search_teacher(request, course_id):
     return my_render_to_response(request, 'yaksh/addteacher.html', context)
 
 
+@login_required
+@email_verified
+def toggle_moderator_role(request):
+    """ Allow moderator to switch to student and back """
+
+    user = request.user
+
+    try:
+        group = Group.objects.get(name='moderator')
+    except Group.DoesNotExist:
+        raise Http404('The Moderator group does not exist')
+
+    if not user.profile.is_moderator:
+        raise Http404('You are not allowed to view this page!')
+
+    if user not in group.user_set.all():
+        group.user_set.add(user)
+    else:
+        group.user_set.remove(user)
```

Comment:
I think the moderator is frequently added or removed from the group is not a good idea!